homeassistant/components/vacuum/__init__.py

```diff
@@ -173,10 +173,12 @@ class _BaseVacuum(Entity):
     Contains common properties and functions for all vacuum devices.
     """
 
+    _attr_supported_features: int
+
     @property
-    def supported_features(self):
+    def supported_features(self) -> int:
         """Flag vacuum cleaner features that are supported."""
-        raise NotImplementedError()
+        return self._attr_supported_features
```

Comment:
Is there a reason that I am not aware of that `vacuum` platform doesn't implement `_attr_supported_features` ?

Reply:
I have implemented short-hand attributes for:
- battery_level
- battery_icon
- fan_speed
- fan_speed_list
- supported_features

Reply:
Don't forget to update the dev docs!

Reply:
Done: https://github.com/home-assistant/developers.home-assistant/pull/1289